### PR TITLE
Complete: Add SmallButton, ChannelForm and ChannelModal components

### PR DESCRIPTION
**Pull request by abhi18av**
#### What does this PR do?
This PR adds a few components developed over the Hackathon.

## Checklist

(Go over all the following points, and put an `x` in all the boxes that apply.)
(If you're unsure about any of these, don't hesitate to ask. We're here to help!)

- [x] My code follows the code style of this project.
- [x] Make sure that you don't have **[WIP]** if you have completed your PR.
- [x] **I am choosing the correct base branch to merge**. 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] And more importantly, I've had fun! :beer:

#### Where should the reviewer start?
(Explain where the reviewer should start in order to review the whole addition or subtractions.)
:checkered_flag:

#### How should this be manually tested?
(List of steps to reproduce, corroborate or tests to run. Write this section clear enough so that external users can also follow it and test the fix.)
1. :one:
2. :two:
3. :tada:

#### Any background context you want to provide?
(Any information regarding the PR that the reviewer should know.)
:construction:

#### What are the relevant tickets?
(Link to issues, related PR, JIRA issues, etc.)
Closes #x

#### Screenshots
(Screenshots of the feature if available.)
:camera:

#### Questions
(If available, to another developer or reviewer.)
:question:

### CODE REVIEW

**Comment by abhi18av** [comment]
This PR should be passing once the https://github.com/pesto-students/batch-8-NuSlack/pull/59 is merged.

**Comment by abhi18av** [comment]
@nimish-gupta this PR adds 3 components which were added as a part of the `hackathon`. For keeping the PR's small I've split the components into other smaller PRs.

File: client/src/components/AddChannelForm/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import axios from 'axios';
+import { Form, Icon, Input, Button, Checkbox } from 'antd';
+
+const AddChannelForm = (props) => {
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    props.form.validateFields(async (err, values) => {
+      if (!err) {
+        console.log('Received values of form: ', values);
```
Comment (nimish-gupta) [inline client/src/components/AddChannelForm/index.js]: Remove console.logs

File: client/src/components/AddChannelForm/index.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            valuePropName: 'checked',
+            initialValue: true,
+          })(<Checkbox>Auto Join</Checkbox>)}
+        </Form.Item>
+      </div>
+      <Form.Item style={{ display: 'flex', justifyContent: 'center' }}>
+        <Button
+          type="primary"
+          htmlType="submit"
+        // className="login-form-button"
```
Comment (nimish-gupta) [inline client/src/components/AddChannelForm/index.js]: Remove comments.

File: client/src/components/AddChannelModal/__test__/__snapshots__/addChannelModal.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (nimish-gupta) [inline client/src/components/AddChannelModal/__test__/__snapshots__/addChannelModal.test.js.snap:1]: Don't need to add the snapshots test.

**Review by nimish-gupta — APPROVED** [review]
You can skip the snapshots tests for frontend. Also for presentational components, you can skip tests.

File: client/src/components/AddChannelForm/index.js (hunk 1 above)
Comment (abhi18av) [reply client/src/components/AddChannelForm/index.js]: Yup, done now.

File: client/src/components/AddChannelForm/index.js (hunk 2 above)
Comment (abhi18av) [reply client/src/components/AddChannelForm/index.js]: Okay, this has been done s well.

File: client/src/components/AddChannelModal/__test__/__snapshots__/addChannelModal.test.js.snap (hunk 3 above)
Comment (abhi18av) [reply client/src/components/AddChannelModal/__test__/__snapshots__/addChannelModal.test.js.snap:1]: Hmm, will keep this in mind.

**Pull request merged**